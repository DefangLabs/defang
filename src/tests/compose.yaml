name: tests
services:
  dfnx:
    build:
      context: .
      dockerfile: Dockerfile
      args:
        DNS: "${COMPOSE_PROJECT_NAME}-dfnx.prod1.defang.dev"
    env_file:
      - fileName.env
    deploy:
      resources:
        limits:
          cpus: '0.50'
          memory: 512M
        reservations:
          cpus: '0.25'
          memory: 256M
    ports:
      - target: 80
        mode: ingress
      - target: 1234
        # mode: host
    secrets:
      - dummy
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost/"]
      # disable: true

  # dfnx:
  #   build:
  #     context: .
      # dockerfile: Dockerfile.dfn
    # ports:
      # - 80

  echo:
    image: ealen/echo-server
    ports:
      - target: 80
        mode: ingress
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost/"]
    # domainname: echotest.gnafed.click
    profiles:
      - donotstart

secrets:
  dummy:
    external: true
    name: dummy
