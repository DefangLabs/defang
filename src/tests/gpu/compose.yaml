services:
  mistral:
    deploy:
      resources:
        reservations:
          devices:
            - capabilities: ["gpu"]
