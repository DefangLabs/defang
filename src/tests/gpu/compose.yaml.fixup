[
  {
    "command": null,
    "deploy": {
      "replicas": 1,
      "resources": {
        "reservations": {
          "devices": [
            {
              "capabilities": [
                "gpu"
              ],
              "count": -1
            }
          ]
        }
      },
      "placement": {}
    },
    "entrypoint": null,
    "networks": {
      "default": null
    }
  }
]