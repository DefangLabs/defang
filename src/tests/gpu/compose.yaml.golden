name: gpu
services:
  mistral:
    deploy:
      resources:
        reservations:
          devices:
            - capabilities:
                - gpu
              count: -1
    networks:
      default: null
networks:
  default:
    name: gpu_default
